src/homepageExperience/components/DataListening.tsx

```diff
@@ -0,0 +1,165 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {withRouter, RouteComponentProps} from 'react-router-dom'
+
+// Apis
+import {runQuery} from 'src/shared/apis/query'
+
+// Components
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import ConnectionInformation, {
+  LoadingState,
+} from 'src/dataLoaders/components/verifyStep/ConnectionInformation'
+
+interface OwnProps {
+  bucket: string
+}
+
+interface State {
+  loading: LoadingState
+  timePassedInSeconds: number
+  secondsLeft: number
+  previousBucket: string
+}
+
+const MINUTE = 60000
+const FETCH_WAIT = 5000
+const SECONDS = 60
+const TIMER_WAIT = 1000
+
+type Props = RouteComponentProps<{orgID: string}> & OwnProps
+
+@ErrorHandling
+class DataListening extends PureComponent<Props, State> {
+  private intervalID: ReturnType<typeof setInterval>
+  private startTime: number
+  private timer: ReturnType<typeof setInterval>
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loading: LoadingState.NotStarted,
```

Comment:
This was mostly copy/paste from the existing file. I just removed the button and triggered the `listen` when a bucket is selected. 

Also re-triggers the check when bucket is changed.